iterators_task_2/student/include/impl.h

```diff
@@ -2,14 +2,46 @@
 #include <iterator>
 
 template <class Container>
-struct accumulator
+struct accumulator : public std::iterator<std::output_iterator_tag, void,void,void,void>
 {
-    /** @todo Iterator traits*/
+
     /** @todo Constructor with container*/
+    accumulator(Container& container) : m_cotainer(container)
+    {}
+
+    accumulator(const accumulator<Container>& other) = default;
+
     /** @todo lvalue writing operator*/
+    template <typename OtherContainer>
+    accumulator& operator=(const OtherContainer& __value) {
```

Comment:
use more descriptive names

Reply:
what do you mean?(should I use "Container" or "Other"?)

Reply:
I mean "value"
actually, value is not a value, it is a container